gdx/src/com/badlogic/gdx/graphics/g2d/ArrayTextureSpriteBatch.java

```diff
@@ -0,0 +1,1604 @@
+		float y1 = transform.m12;
+		float x2 = transform.m01 * height + transform.m02;
+		float y2 = transform.m11 * height + transform.m12;
+		float x3 = transform.m00 * width + transform.m01 * height + transform.m02;
+		float y3 = transform.m10 * width + transform.m11 * height + transform.m12;
+		float x4 = transform.m00 * width + transform.m02;
+		float y4 = transform.m10 * width + transform.m12;
+
+		float u = region.u * subImageScaleWidth;
+		float v = region.v2 * subImageScaleHeight;
+		float u2 = region.u2 * subImageScaleWidth;
+		float v2 = region.v * subImageScaleHeight;
+
+		float color = this.colorPacked;
+
+		vertices[idx++] = x1;
+		vertices[idx++] = y1;
+		vertices[idx++] = color;
+		vertices[idx++] = u;
+		vertices[idx++] = v;
+		vertices[idx++] = ti;
+
+		vertices[idx++] = x2;
+		vertices[idx++] = y2;
+		vertices[idx++] = color;
+		vertices[idx++] = u;
+		vertices[idx++] = v2;
+		vertices[idx++] = ti;
+
+		vertices[idx++] = x3;
+		vertices[idx++] = y3;
+		vertices[idx++] = color;
+		vertices[idx++] = u2;
+		vertices[idx++] = v2;
+		vertices[idx++] = ti;
+
+		vertices[idx++] = x4;
+		vertices[idx++] = y4;
+		vertices[idx++] = color;
+		vertices[idx++] = u2;
+		vertices[idx++] = v;
+		vertices[idx++] = ti;
+	}
+
+	/** Convenience method to flush if the Batches vertex-array cannot hold an additional sprite ((spriteVertexSize + 1) * 4
+	 * vertices) anymore. */
+	private void flushIfFull () {
+		// original Sprite attribute size plus one extra float per sprite vertex
+		if (vertices.length - idx < spriteFloatSize + spriteFloatSize / spriteVertexSize) {
+			flush();
+		}
+	}
+
+	@Override
+	public void flush () {
+
+		if (idx == 0) return;
+
+		renderCalls++;
+		totalRenderCalls++;
+
+		int spritesInBatch = idx / (spriteFloatSize + 4);
+		if (spritesInBatch > maxSpritesInBatch) maxSpritesInBatch = spritesInBatch;
+		int count = spritesInBatch * 6;
+
+		if (useMipMaps && mipMapsDirty) {
+			Gdx.gl30.glGenerateMipmap(GL30.GL_TEXTURE_2D_ARRAY);
+			mipMapsDirty = false;
+		}
+
+		Mesh mesh = this.mesh;
+
+		mesh.setVertices(vertices, 0, idx);
+
+		mesh.getIndicesBuffer().position(0);
+		mesh.getIndicesBuffer().limit(count);
+
+		if (blendingDisabled) {
+			Gdx.gl.glDisable(GL30.GL_BLEND);
+		} else {
+			Gdx.gl.glEnable(GL30.GL_BLEND);
+			if (blendSrcFunc != -1) Gdx.gl.glBlendFuncSeparate(blendSrcFunc, blendDstFunc, blendSrcFuncAlpha, blendDstFuncAlpha);
+		}
+
+		if (customShader != null) {
+			mesh.render(customShader, GL30.GL_TRIANGLES, 0, count);
+		} else {
+			mesh.render(shader, GL30.GL_TRIANGLES, 0, count);
+		}
+
+		idx = 0;
+	}
+
+	/** Assigns space on the Array Texture, sets up Texture scaling and manages the LFU cache.
+	 * @param texture The texture that shall be loaded into the cache, if it is not already loaded.
+	 * @return The texture slot that has been allocated to the selected texture */
+	private int activateTexture (Texture texture) {
+
+		subImageScaleWidth = texture.getWidth() * invMaxTextureWidth;
```

Comment:
Tested caching the last active texture, which reduces checking times. At least with map rendering it made performance better (drawing the same texture several times  in a row)

```
//Add private cache
private Texture lastTexture;
private int lastTextureIndex = -1;

flush(){ 
              lastTexture = null;
}
begin (){
              lastTexture = null;
              }

private int activateTextureCached(Texture texture){
        if(lastTexture != null && lastTexture.equals(texture)){
            return lastTextureIndex;
        }else {
            lastTexture = texture;
            return lastTextureIndex = activateTexture(texture);
        }
    }
```

Reply:
If you don't specifically need the behavior of .equals(), are fine with reference equality of textures, and know the parameter passed to activateTextureCached() is not null, you might be able to just skip the null check and do `if(lastTexture == texture)`. I have no idea what Texture.equals() does internally, though it might just be a reference equality check anyway.